pkg/inject/oauth_proxy.go

```diff
@@ -97,3 +98,35 @@ func getOAuthProxyContainer(jaeger *v1.Jaeger) corev1.Container {
 		Resources: commonSpec.Resources,
 	}
 }
+
+//PreserveOauthCookieSecret preserve the generated oauth cookie across multiple reconciliations
+func PreserveOauthCookieSecret(specSrc, specDst *appsv1.DeploymentSpec) {
```

Comment:
This could have a better name. Perhaps `PropagateOAuthCookieSecret`? Or just `HandleOAuthCookieSecret`?

Reply:
In recent changes, I've been trying to avoid those pointers, returning the resulting spec. You could also do it, so that your caller would look like this:

```
tp.Spec = inject.PropagateOAuthCookieSecret(&t.Spec, &tp.Spec)
```

Reply:
Sounds reasonable. I would prefer to not modify funcion parameters , make it pure function.

About the names, I'll rename it. I just put a the first name came to my mind just to test if this  fix the issue.

Thanks.